text/0148-sampling-probability.md

```diff
@@ -0,0 +1,744 @@
+
+A metrics processor can be configured to limit cardinality for a
+single metric name, allowing no more than K distinct label sets per
+export interval.  The export interval is fixed to a short interval so
+that a complete set of distinct labels can be stored temporarily.
+
+Caveats: as presented, this works for Sum and Histogram points
+received with Delta aggregation temporality and where the Sum is
+monotonic (see
+[opentelemetry-proto/issues/303](https://github.com/open-telemetry/opentelemetry-proto/issues/303)).
+
+Considering data points received during the interval, when the number
+of points exceeds K, select a probability proportional to size sample
+of points, output every point with an additional (non-descriptive)
+`sampling.cardinality_limit.adjusted_count` attribute.
+</details>
+
+## Explanation
+
+Consider a hypothetical telemetry signal in which an API event
+produces a unit of data that has one or more associated numbers.
+Using the OpenTelemetry Metrics data model terminology, we have two
+scenarios in which sampling is common.
+
+1. _Counter events:_ Each event represents a count, signifying the change in a sum.
+2. _Histogram events:_ Each event represents an individual variable, signifying membership in a distribution.
+
+A Tracing Span event qualifies as both of these cases simultaneously.
+One span can be interpreted as at least one Counter event (e.g., one
+request, the number of bytes read) and at least one Histogram event
+(e.g., request latency, request size).
+
+In Metrics, [Statsd Counter and Histogram events meet this definition](https://github.com/statsd/statsd/blob/master/docs/metric_types.md#sampling).
+
+In both cases, the goal in sampling is to estimate the count of events
+in the whole population, meaning all the events, using only the events
+that were selected in the sample.
+
+### Model and terminology
+
+This model is meant to apply in telemetry collection situations where
+individual events at an API boundary are sampled for collection.  Once
+the process of sampling individual API-level events is understood, we
+will learn to apply these techniques for sampling aggregated data.
+
+In sampling, the term _sampling design_ refers to how sampling
+probability is decided and the term _sample frame_ refers to how
+events are organized into discrete populations.
+
+For example, a simple design uses uniform probability, and a simple
+framing technique is to collect one sample per distinct span name.
+
+After executing a sampling design over a frame, each item selected in
+the sample will have known _inclusion probability_, that determines
+how likely the item was to being selected.  Implicitly, all the items
+that were not selected for the sample have zero inclusion probability.
+
+Descriptive words that are often used to describe sampling designs:
+
+- *Fixed*: the sampling design is the same from one frame to the next
+- *Adaptive*: the sampling design changes from one frame to the next
```

Comment:
I understand that these are hard to define.
Just curious if we want to consider something like "Adaptive: the sampling design changes from one frame to the next based on the input values". And here goes the reason:

If the sampling design changes dynamically based on the configuration from the operator, it sounds like it is "not fixed" but also "not adaptive".

Reply:
I added "based on the observed data".